File: 프로그래머스/0/120878. 유한소수 판별하기/유한소수 판별하기.py
import math as m

def solution(a, b):
    answer = 1 
    if ((a//m.gcd(a,b)) / (b//m.gcd(a,b))) % 1 == 0:
        answer = 1
    else:
        for i in range(2, b//m.gcd(a,b) + 1 ):
            if (b//m.gcd(a,b)) % i == 0 and i % 2 != 0 and i % 5 != 0:
                answer = 2
                break
    return answer

File: 프로그래머스/0/120878. 유한소수 판별하기/유한소수 판별하기.java
class Solution {
    public int solution(int a, int b) {
    int x = b/gcd(a,b);
    while (x % 2 == 0 || x % 5 == 0){
        if (x % 2 == 0){
            x /= 2;
        }

        else if (x % 5 == 0){
            x /= 5; 
        }
    }

    if (x == 1){
        return 1;
    }

    return 2;
}

public static int gcd(int a, int b){
    if(b==0) return a;
    return gcd(b,a%b);
}
}